core/src/main/java/org/apache/iceberg/BaseRewriteManifests.java

```diff
@@ -275,14 +275,17 @@ private boolean matchesPredicate(ManifestFile manifest) {
     return predicate == null || predicate.test(manifest);
   }
 
-  private void validateDeletedManifests(Set<ManifestFile> currentManifests) {
+  private void validateDeletedManifests(
+      Set<ManifestFile> currentManifests, long currentSnapshotID) {
     // directly deleted manifests must be still present in the current snapshot
     deletedManifests.stream()
         .filter(manifest -> !currentManifests.contains(manifest))
         .findAny()
         .ifPresent(
             manifest -> {
-              throw new ValidationException("Manifest is missing: %s", manifest.path());
+              throw new ValidationException(
+                  "Deleted manifest %s could not be found in the current snapshot %d",
```

Comment:
Nit: Could we just say "Deleted manifest %s could not be found in snapshot %d". Sorry I think I said "current" in Slack,  but I actually think it may confuse users because by the time it surfaces the "current" can be different than what's in the message.

Reply:
"current" is the "latest". So, I feel it is ok as we are printing snapshot id